django/db/backends/base/schema.py

```diff
@@ -98,6 +98,10 @@ def execute(self, sql, params=()):
         """
         Executes the given SQL statement, with optional parameters.
         """
+        if not self.collect_sql and self.connection.in_atomic_block and not self.connection.features.can_rollback_ddl:
```

Comment:
Do not perform the check when collecting the SQL as nothing is going to be executed anyway. This allows the SQL to be collected while in a transaction which is something `model_options.test_tablespaces.TablespacesTests` relies on.